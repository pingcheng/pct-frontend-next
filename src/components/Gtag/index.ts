import Gtag from "./Gtag";

export default Gtag;
