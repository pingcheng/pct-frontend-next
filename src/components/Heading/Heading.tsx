import styles from "./style.module.scss";

type HeadingProps = {
  text: string;
  align?: "left" | "center" | "right";
};

export function Heading({ text, align = "left" }: HeadingProps) {
  const classLists = [styles.heading1];

  switch (align) {
    case "center":
      classLists.push(styles.center);
      break;
    case "right":
      classLists.push(styles.right);
      break;
    default:
  }

  return <h1 className={classLists.join(" ")}>{text}</h1>;
}
