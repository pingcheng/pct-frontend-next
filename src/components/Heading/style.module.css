@reference "../../app/globals.css";

.heading1 {
  @apply text-3xl font-extrabold mb-4 tracking-tight;
  color: var(--color-text-light);
  position: relative;
  padding-bottom: 0.25em;
}

.heading1::after {
  content: "";
  display: block;
  width: 2.5rem;
  height: 0.18em;
  background: var(--color-accent);
  border-radius: 1em;
  margin: 0.5em auto 0 auto;
}

.center {
  text-align: center;
}

.right {
  text-align: right;
}

@media (prefers-color-scheme: dark) {
  .heading1 {
    color: var(--color-text-dark);
  }
}
