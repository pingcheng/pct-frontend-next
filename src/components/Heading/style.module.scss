.heading1 {
  @apply text-2xl;
}

.center {
  text-align: center;
}

.right {
  text-align: right;
}