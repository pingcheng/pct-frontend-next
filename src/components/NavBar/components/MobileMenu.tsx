"use client";

import Link from "next/link";
import { usePathname } from "next/navigation";
import styles from "../style.module.css";
import { forwardRef } from "react";

type MenuItem = {
  label: string;
  path: string;
  activePattern?: RegExp;
};

type MobileMenuProps = {
  menuItems: MenuItem[];
  isVisible: boolean;
  onItemClick: () => void;
};

function isActive(item: MenuItem, pathName: string): boolean {
  if (item.activePattern) {
    return item.activePattern.test(pathName);
  }
  return pathName === item.path;
}

export const MobileMenu = forwardRef<HTMLDivElement, MobileMenuProps>(
  ({ menuItems, isVisible, onItemClick }, ref) => {
    const pathName = usePathname();

    return (
      <div
        id="mobile-menu"
        ref={ref}
        className={`sm:hidden ${styles.mobileMenu} ${
          isVisible ? styles.expanded : ""
        }`}
        aria-hidden={!isVisible}
      >
        <div className="px-4 pt-2 pb-3 space-y-1">
          {menuItems.map((item) => (
            <Link
              key={item.path}
              href={item.path}
              className={`${styles.navItem} ${
                isActive(item, pathName) ? styles["active"] : ""
              }`}
              onClick={onItemClick}
            >
              {item.label}
            </Link>
          ))}
        </div>
      </div>
    );
  }
);

MobileMenu.displayName = "MobileMenu";