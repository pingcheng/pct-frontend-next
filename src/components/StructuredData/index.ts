export * from "./StructuredData";
export * from "./PersonStructuredDataProvider";
