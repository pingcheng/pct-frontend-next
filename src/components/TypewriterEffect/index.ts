export { TypewriterEffect } from "./TypewriterEffect";
export { CyclingTypewriterEffect } from "./CyclingTypewriterEffect";