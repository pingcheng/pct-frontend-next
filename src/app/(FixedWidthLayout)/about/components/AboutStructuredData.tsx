import {
  BreadcrumbListStructuredData,
  ProfessionalServiceStructuredData,
} from "@/components/StructuredData/StructuredData";
import { PersonStructuredDataProvider } from "@/components/StructuredData/PersonStructuredDataProvider";
import { profile } from "@/data/profile";
import { backendStack, devOpsStack, frontendStack, legacyStack } from "@/data/skills";
import { URLS } from "@/constants/urls";

export function AboutStructuredData() {
  const skillsArray = [
    ...backendStack,
    ...frontendStack,
    ...devOpsStack,
    ...legacyStack,
  ];

  const breadcrumbItems = [
    { name: "Home", url: URLS.WEBSITE.BASE },
    { name: "About", url: URLS.WEBSITE.ABOUT },
  ];

  return (
    <>
      <PersonStructuredDataProvider 
        knowsAbout={skillsArray}
        includeEmail={true}
      />
      <BreadcrumbListStructuredData items={breadcrumbItems} />
      <ProfessionalServiceStructuredData
        name="Software Development Services"
        description="Full-stack web development, cloud solutions, and enterprise software development services specializing in Node.js, TypeScript, PHP, Java, and modern web technologies."
        provider={{
          name: profile.fullName,
          url: URLS.WEBSITE.BASE,
        }}
        areaServed="Melbourne, Australia"
        serviceType="Software Development"
      />
    </>
  );
}