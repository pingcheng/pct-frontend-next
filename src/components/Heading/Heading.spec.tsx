import { render, screen } from "@testing-library/react";
import { Heading } from "@/components/Heading";

describe("test <Heading />", () => {
  it("should match snapshot", () => {
    const { container } = render(<Heading text="Test Heading" />);
    expect(container).toMatchSnapshot();
  });

  it("should render the provided text", () => {
    render(<Heading text="Test Heading" />);
    expect(screen.getByText("Test Heading")).toBeInTheDocument();
  });

  it("should have default left alignment", () => {
    render(<Heading text="Test Heading" />);
    const heading = screen.getByRole("heading", { level: 1 });
    expect(heading).not.toHaveClass("center");
    expect(heading).not.toHaveClass("right");
  });

  it("should apply center alignment when specified", () => {
    render(<Heading text="Test Heading" align="center" />);
    const heading = screen.getByRole("heading", { level: 1 });
    expect(heading).toHaveClass("center");
  });

  it("should apply right alignment when specified", () => {
    render(<Heading text="Test Heading" align="right" />);
    const heading = screen.getByRole("heading", { level: 1 });
    expect(heading).toHaveClass("right");
  });

  it("should apply custom className when provided", () => {
    render(<Heading text="Test Heading" className="custom-class" />);
    const heading = screen.getByRole("heading", { level: 1 });
    expect(heading).toHaveClass("custom-class");
  });
});
