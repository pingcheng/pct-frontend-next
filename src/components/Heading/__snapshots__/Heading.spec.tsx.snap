// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test <Heading /> should match snapshot 1`] = `
<div>
  <h1
    class="heading1"
  >
    Test Heading
  </h1>
</div>
`;
